feat(Auth): Adding documentation for TOTP support (#5690)
aws-amplify/docs: src/fragments/lib/auth/common/mfa_preferences/common.mdx

<Callout>

> Fetching and Updating MFA preferences is only available through the escape hatch APIs under the AWS Cognito Auth plugin. 

Reviewer: I thought the escape hatch was different. Should this say:

```suggestion
The APIs for fetching and updating MFA preferences are only available on the AWS Cognito Auth plugin and not on the Auth category.
```

Author: Updated.

</Callout>

> 

Reviewer 2: can we add an intro here about what TOTP is, and why a customer would use it before the callout.

You can use Time-based One-Time Password (TOTP) for multi-factor authentication (MFA) in your web or mobile applications. Amplify's Auth category includes support for handling TOTP setup and codes, offering an integrated solution for enhanced security for your users. Authenticator apps generate the temporary, one-time passwords. These apps, such as Google Authenticator, Microsoft Authenticator, have the TOTP algorithm built-in and work by using a shared secret key and the current time to generate a short-lived password.

Author: Done.

## Fetch the current user's MFA preferences

> Invoke the following api to get the MFA preferences for the user.

Reviewer: ```suggestion
Invoke the following API to get the MFA preference for the current user.
```

I would also change `get` to `fetch` or `retrieve`

Author: Updated.

import iosFetchMFAPreference from "/src/fragments/lib/auth/ios/mfa_preference/10_fetch_mfa_preference.mdx";

<Fragments fragments={{ios: iosFetchMFAPreference}} />

## Update the current user's MFA preferences

> Invoke the following api to update the MFA preferences for the user.

Author: Updated.

> <Callout>

Author: Agreed. Updated.

> Only one MFA method can be marked as preferred at a time. If the user has multiple MFA methods enabled, and tries to mark more than one MFA method as preferred, the API will throw an error.

Reviewer: ```suggestion
Only one MFA method can be marked as preferred at a time. If the user has multiple MFA methods enabled and tries to mark more than one MFA method as preferred, the API will throw an error.
```

Author: Updated.

</Callout>

import iosUpdateMFAPreferences from "/src/fragments/lib/auth/ios/mfa_preference/20_update_mfa_preference.mdx";

<Fragments fragments={{ios: iosUpdateMFAPreferences}} />

> **Note:** If multiple MFA methods are enabled for the user, the `signIn` API will return `continueSignInWithMFASelection` as the next step in the auth flow. During this scenario, the user should be prompted to select the MFA method they want to use to sign in.

Reviewer: ```suggestion
If multiple MFA methods are enabled for the user, the `signIn` API will return `continueSignInWithMFASelection` as the next step in the auth flow. During this scenario, the user should be prompted to select the MFA method they want to use to sign in.
```

Should we link to the sign-in steps doc? Or provide a code sample here?

Author: Good call out.. Let me link it to sign in steps.

Author: Updated/
